### PR TITLE
feat(2319): Add template order docs

### PR DESCRIPTION
**Pull request by tkyi**
## Context

Users can now use templates in templates. They can also use the new field `order` to pick and choose steps from a template and their own job configs.

## Objective

This PR adds docs for the new feature.

## References

Related to https://github.com/screwdriver-cd/screwdriver/issues/2319

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by vnugopal — APPROVED** [review]

File: docs/user-guide/configuration/jobconfiguration.md [hunk 1]
```diff
@@ -111,15 +115,63 @@ jobs:
...
     main2:
         requires: [main]
         image: node:8
         steps:
-            - greet: echo hello
+            - test: echo Skipping test
+
+```
+
+## Merge shared steps
```
Comment (jithine) [inline docs/user-guide/configuration/jobconfiguration.md]: `mergeSharedSteps` is applicable only when using template

File: docs/user-guide/templates.md [hunk 2]
```diff
@@ -168,6 +173,63 @@ jobs:
 
 ```
 
+### Step order
```
Comment (jithine) [inline docs/user-guide/templates.md:228]: Main heading should be `Template composition/inheritance` `Order` then being a sub-heading under it

Comment (tkyi) [reply docs/user-guide/templates.md:228]: I saw this as something else to go under `Overriding Template Steps`, but I can move it

File: docs/user-guide/configuration/jobconfiguration.md (hunk 1 above)
Comment (tkyi) [reply docs/user-guide/configuration/jobconfiguration.md]: removed

**Review by adong — APPROVED** [review]

**Pull request merged**